src/gpodder/plugins/itunes.py

```diff
@@ -27,38 +27,41 @@
 
 logger = logging.getLogger(__name__)
 
-ITUNES_DEFAULT_VERSION = '11.1.5'
-
-ITUNES_FEEDURL_RE = {'10.7': r'feed-url="([^"]+)"',
-                     '11.1.5': r'"feedUrl":\s*"([^"]+)"'}
-
-
 class ITunesFeedException(BaseException):
     pass
 
 
 @registry.feed_handler.register
 def itunes_feed_handler(channel, max_episodes, config):
-    m = re.match(r'https?://(podcasts|itunes).apple.com/(?:[^/]*/)?podcast/.+$', channel.url, re.I)
+    m = re.match(r'https?://(podcasts|itunes)\.apple\.com/(?:[^/]*/)?podcast/.*id(?P<podcast_id>[0-9]+).*$', channel.url, re.I)
     if m is None:
         return None
 
     logger.debug('Detected iTunes feed.')
-    version = ITUNES_DEFAULT_VERSION
-    headers = {'User-agent': 'iTunes/{}'.format(version)}
+
+    if not m.group('podcast_id').isnumeric or int(m.group('podcast_id')) < 0:
```

Comment:
`str.isnumeric` is a method that you would need to call (so `m.group('podcast_id').isnumeric()`), otherwise it will always be `true`-ish` in a boolean context. But as @elelay said, it's already part of the regex above, so it's guaranteed to be just digits 0-9 (one or more), so I'd just remove this and the negative integer check as well, as the regex guarantees that it will be a positive integer string.